Add a very simple tool for recording system diagnosis info.

130s opened the pull request:
Not sure if this type of tools should go into this package.

A tool that the users who are reporting issues can simply use with just a single line like below:

```
$ curl -sL https://raw.githubusercontent.com/%REPOSITORY%diagnosis_info.sh | bash -
$ curl -sL https://raw.githubusercontent.com/130s/rtmros_common/add/diagnosiscommand/hrpsys_tools/scripts/diagnosis_info.sh | bash -    (Example)
```

Or if we decide to put this script in this package, we can `rosrun`:
```
$ rosrun hrpsys_tools diagnosis_info.sh
```

The output is saved under `/tmp`, which is supposed to be printed when the script finishes:

> *** All diagnostic info recorded. Ask a question at rtm-ros-robotics@googlegroups.com, with /tmp/rtmros_diagnosisinfo_commands_20150122-052151.log file, and /tmp/rtmros_diagnosisinfo_ros_20150122-052151.tgz too if you're using ROS.

130s commented:
Interest for this type of tool [was mentioned here](https://github.com/start-jsk/rtmros_common/pull/333#issuecomment-36254808).

130s commented on hrpsys_tools/scripts/diagnosis_info.sh:
```diff
@@ -0,0 +1,21 @@
+
+env |grep ROS | tee -a ${FILENAME_LOG_COMMANDS}
+ifconfig | tee -a ${FILENAME_LOG_COMMANDS}
+rosrun rtshell rtls %HOST_ROBOT%:15005/  | tee -a ${FILENAME_LOG_COMMANDS}
```
Need to accept host info as an argument.

130s commented on hrpsys_tools/scripts/diagnosis_info.sh:
```diff
@@ -0,0 +1,21 @@
+
+tar cfvz ${FILENAME_LOG_ALL} ${FILENAME_LOG_COMMANDS} ${FILENAME_LOG_ROS}
+
+echo "*** All diagnostic info recorded into a tarball: ${FILENAME_LOG_ALL}"
```
Need to print what this script exactly does. Otherwise the user has no idea what this does, might even refrain from using it from security point of view.

k-okada replied:
show about 
```
aptitude -q -F "%?p=%?V %M" --disable-columns search 'ros-*-(hrp|rtm)'
```
tools in this repo should be (at least it looks like) general

--
◉ Kei Okada


On Fri, Jan 23, 2015 at 8:27 AM, Isaac I.Y. Saito <notifications@github.com> 
wrote:

> In hrpsys_tools/scripts/diagnosis_info.sh 
> <https://github.com/start-jsk/rtmros_common/pull/636#discussion_r23420143>
> :
>
> > +FILENAME_LOG_ALL=/tmp/rtmros_diagnosisinfo_all_`date +"%Y%m%d-%H%M%S"`.tgz
> > +
> > +env |grep ROS | tee -a ${FILENAME_LOG_COMMANDS}
> > +ifconfig | tee -a ${FILENAME_LOG_COMMANDS}
> > +rosrun rtshell rtls %HOST_ROBOT%:15005/  | tee -a ${FILENAME_LOG_COMMANDS}
> > +dpkg -p ros-hydro-hironx-ros-bridge | grep Ver  | tee -a ${FILENAME_LOG_COMMANDS}
> > +dpkg -p ros-hydro-rtmros-common | grep Ver  | tee -a ${FILENAME_LOG_COMMANDS}
> > +dpkg -p ros-hydro-hrpsys | grep Ver  | tee -a ${FILENAME_LOG_COMMANDS}
> > +dpkg -p ros-hydro-openhrp3 | grep Ver  | tee -a ${FILENAME_LOG_COMMANDS}
> > +dpkg -p ros-hydro-openrtm-aist-core | grep Ver  | tee -a ${FILENAME_LOG_COMMANDS}
> > +
> > +tar -C ~/.ros/log -cvzf ${FILENAME_LOG_ROS} `cd ~/.ros/log; ls -d * | head -1`
> > +
> > +tar cfvz ${FILENAME_LOG_ALL} ${FILENAME_LOG_COMMANDS} ${FILENAME_LOG_ROS}
> > +
> > +echo "*** All diagnostic info recorded into a tarball: ${FILENAME_LOG_ALL}"
>
> Need to print what this script exactly does. Otherwise the user has no 
> idea what this does, might even refrain from using it from security point 
> of view.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/start-jsk/rtmros_common/pull/636/files#r23420143>.
>

130s replied:
> aptitude -q -F "%?p=%?V %M" --disable-columns search 'ros-*-(hrp|rtm)'

Very nice! but `aptitude` isn't installed by default. I'm trying to find alternative.

130s commented:
Done. 

> Very nice! but aptitude isn't installed by default. I'm trying to find alternative.

In https://github.com/130s/rtmros_common/commit/9ea3ece1731e93c3000c682b3db622e4cda39192 I added:

```
dpkg -l | grep '^ii' | grep ros- | awk '{print $2 "\t" $3}' 2>&1 | tee -a ${FILENAME_LOG_COMMANDS}
```
This may potentially include garbage though.

Also, to show the respect toward those early ROS developers, I renamed the file. Now it can be run by:

```
$ rosrun hrpsys_tools rtmroswtf.sh
$ rosrun hrpsys_tools rtmroswtf.sh hiro016 15004
```

k-okada commented:
nice, a like both idea!

--
◉ Kei Okada


On Sat, Jan 31, 2015 at 11:58 AM, Isaac I.Y. Saito <notifications@github.com
> wrote:

> Done. 
>
> Very nice! but aptitude isn't installed by default. I'm trying to find 
> alternative.
>
> In 130s@9ea3ece 
> <https://github.com/130s/rtmros_common/commit/9ea3ece1731e93c3000c682b3db622e4cda39192> 
> I added:
>
> dpkg -l | grep '^ii' | grep ros- | awk '{print $2 "\t" $3}' 2>&1 | tee -a ${FILENAME_LOG_COMMANDS}
>
> This may potentially include garbage though.
>
> Also, to show the respect toward those early ROS developers, I renamed the 
> file. Now it can be run by:
>
> $ rosrun hrpsys_tools rtmroswtf.sh
> $ rosrun hrpsys_tools rtmroswtf.sh hiro016 15004
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/start-jsk/rtmros_common/pull/636#issuecomment-72301548>
> .
>

130s commented:
rtmrosawesome
